### PR TITLE
HBASE-27362 CompactSplit.requestCompactionInternal may bypass compact…

### PR DESCRIPTION
**Pull request by comnetwork**
…ionsEnabled check

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 59s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 26s |  master passed  |
| +1 :green_heart: |  compile  |   2m 15s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 41s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  8s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   7m 58s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 39s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  28m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 3838dc595340 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d2cc840edf |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  master passed  |
| +1 :green_heart: |  compile  |   0m 43s |  master passed  |
| -1 :x: |  shadedjars  |   0m  8s |  branch has 7 errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 192m 58s |  hbase-server in the patch failed.  |
|  |   | 206m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3b6a54a40045 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d2cc840edf |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| shadedjars | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/artifact/yetus-jdk11-hadoop3-check/output/branch-shadedjars.txt |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/testReport/ |
| Max. process+thread count | 2821 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m 52s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 29s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 46s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 219m  8s |  hbase-server in the patch passed.  |
|  |   | 243m  4s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0f8075606420 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d2cc840edf |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/testReport/ |
| Max. process+thread count | 2627 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java [hunk 1]
```diff
@@ -229,8 +229,10 @@ private synchronized void requestSplit(final Region r, byte[] midKey) {
    */
   private synchronized void requestSplit(final Region r, byte[] midKey, User user) {
     if (midKey == null) {
-      LOG.debug("Region " + r.getRegionInfo().getRegionNameAsString()
-        + " not splittable because midkey=null");
+      if (LOG.isDebugEnabled()) {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:232]: Are these logging changes related?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java [hunk 2]
```diff
@@ -312,14 +316,20 @@ public void switchCompaction(boolean onOrOff) {
     if (onOrOff) {
       // re-create executor pool if compactions are disabled.
       if (!isCompactionsEnabled()) {
-        LOG.info("Re-Initializing compactions because user switched on compactions");
+        if (LOG.isInfoEnabled()) {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:319]: I prefer we keep the old style.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java [hunk 3]
```diff
@@ -336,6 +346,13 @@ private void requestCompactionInternal(HRegion region, String why, int priority,
   protected void requestCompactionInternal(HRegion region, HStore store, String why, int priority,
     boolean selectNow, CompactionLifeCycleTracker tracker,
     CompactionCompleteTracker completeTracker, User user) throws IOException {
+    if (!this.isCompactionsEnabled()) {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:349]: So this is the actual fix?

Comment (comnetwork) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:349]: yes, it is the actual fix.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java (hunk 1 above)
Comment (comnetwork) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:232]: no, it is not related, just modified by the way.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java (hunk 2 above)
Comment (comnetwork) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:319]: In fact in this class ,some logs using the {{LOG.isInfoEnabled}} or {{LOG.isDebugEnabled}}, and other logs using {{LOG.info}} or {{LOG.debug}} directly. If you do not like this, I would revert it.

Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java]: If we  want to align them, I prefer we open a new issue.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 24s |  master passed  |
| +1 :green_heart: |  compile  |   2m 16s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 41s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 10s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 18s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   7m 52s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 38s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  28m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 97e34b50b6f2 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fa3bd7feb9 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java [hunk 4]
```diff
@@ -315,11 +315,13 @@ public void switchCompaction(boolean onOrOff) {
         LOG.info("Re-Initializing compactions because user switched on compactions");
         reInitializeCompactionsExecutors();
       }
-    } else {
-      LOG.info("Interrupting running compactions because user switched off compactions");
-      interrupt();
+      setCompactionsEnabled(onOrOff);
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:318]: Let's also revert this change? It does not change anything actually? Only a style change?

Comment (comnetwork) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java:318]: Oh, here I think it is not just a style change, for disable compaction, I think the order we would better to set the flag first and then shut down the thread pool, in the reverse order of operations when compaction is enabled.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  7s |  master passed  |
| +1 :green_heart: |  compile  |   0m 55s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 273m 14s |  hbase-server in the patch passed.  |
|  |   | 297m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 214deb71d4a7 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fa3bd7feb9 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/testReport/ |
| Max. process+thread count | 2626 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 44s |  master passed  |
| +1 :green_heart: |  compile  |   0m 42s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 284m 12s |  hbase-server in the patch failed.  |
|  |   | 306m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4768 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 78916d8cb77a 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fa3bd7feb9 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/testReport/ |
| Max. process+thread count | 2596 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4768/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Comment by ndimiduk** [comment]
Nice one 👍
